[build] Add editable mode unittest

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #8816
* __->__ #8817

As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

larryliu0820 committed:
[build] Add editable mode unittest

As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[build] Add editable mode unittest"


As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[build] Add editable mode unittest"


As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[build] Add editable mode unittest"


As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

[ghstack-poisoned]

mergennachin commented:
Thanks for adding this test, looks like a good change, let's fix CI first.

swolchok commented on .ci/scripts/utils.sh:
```diff
@@ -166,3 +162,60 @@ do_not_use_nightly_on_ci() {
 }
+
+
+parse_args() {
```
this is reaching the complexity level when I start wondering if it's easier overall to switch from a shell script to python. (not blocking, out of scope for this change)

swolchok commented on .ci/scripts/setup-linux.sh:
```diff
@@ -10,19 +10,17 @@ set -exu
-install_executorch "use-pt-pinned-commit"
-build_executorch_runner "${BUILD_TOOL}" "${2:-Release}"
+if [[ "$EDITABLE" == "true" ]]; then
+  install_executorch "--use-pt-pinned-commit" "--editable"
+else
+  install_executorch "--use-pt-pinned-commit"
```
nit: no need to quote the flags here, it's unusual to do so

swolchok commented on .github/workflows/_unittest.yml:
```diff
@@ -15,6 +15,10 @@ on:
+      editable:
+        required: false
+        type: string
+        description: Install ExecuTorch in editable or not.
```
nit: s/editable or not/editable mode or not/

swolchok reviewed (approved):
if the tests pass, ship it

larryliu0820 committed:
Update on "[build] Add editable mode unittest"


As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[build] Add editable mode unittest"


As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[build] Add editable mode unittest"


As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[build] Add editable mode unittest"


As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[build] Add editable mode unittest"


As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[build] Add editable mode unittest"


As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[build] Add editable mode unittest"


As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[build] Add editable mode unittest"


As titled. To do this we need to refactor the arguments being passed
into `setup-linux.sh` `setup-macos.sh` and `unittest-linux.sh`
`unittest-macos.sh`.

[ghstack-poisoned]
